addons/mass_mailing/tests/test_mailing_internals.py

```diff
@@ -313,6 +313,68 @@ def test_mailing_computed_fields_default(self):
         })
         self.assertEqual(literal_eval(mailing.mailing_domain), [('email', 'ilike', 'test.example.com')])
 
+    @users('user_marketing')
+    def test_mailing_computed_fields_default_email_from(self):
+        # Testing if the email_from is correctly computed when an
+        # alias domain for the company is set
+
+        # Setup mail outgoing server for use cases
+        # Case where alias domain is set and there is a default outgoing email server
+        # for mass mailing. from_filter matches domain of company alias domain (mycompany.com)
+        # before record creation
+        from_filter_match = self.env['ir.mail_server'].sudo().create({
+                'name' : 'mass_mailing_test_match_from_filter',
+                'from_filter' : 'test.mycompany.com',
+                'smtp_host' : 'not_real@smtp.com',
+            })
+
+        # Case where alias domain is set and there is a default outgoing email server
+        # for mass mailing. from_filter DOES NOT match domain of company alias domain
+        # before record creation
+        from_filter_missmatch = self.env['ir.mail_server'].sudo().create({
+                'name' : 'mass_mailing_test_from_missmatch',
+                'from_filter' : 'test.com',
+                'smtp_host' : 'not_real@smtp.com',
+            })
```

Comment:
Can be created in multi (and use variables for alias domain), like
```suggestion
        from_filter_match, from_filter_missmatch = self.env['ir.mail_server'].sudo().create([
            # Case where alias domain is set and there is a default outgoing email server
            # for mass mailing. from_filter matches domain of company alias domain (mycompany.com)
            # before record creation
            {
                'name' : 'mass_mailing_test_match_from_filter',
                'from_filter' : self.alias_domain,
                'smtp_host' : 'not_real@smtp.com',
            },
            # Case2 where alias domain is set and there is a default outgoing email server
            # for mass mailing. from_filter DOES NOT match domain of company alias domain
            # before record creation
            {
                'name' : 'mass_mailing_test_from_missmatch',
                'from_filter' : 'test.com',
                'smtp_host' : 'not_real@smtp.com',
            },
        ])
```